### PR TITLE
fix lldb debugger for ubuntu distros

### PR DESCRIPTION
**Pull request by TomWeaver18**
Add out the box support for lldb on ubuntu distros.

Have run the lint and unit tests.

Again, python not my strong suit, appreciate any feedback on this.

I avoided using the python 3,x approach as wasn't sure if it would be supported on both 2.7 and 3.x.

happy reviewing
Tom

### CODE REVIEW

**Comment by TomWeaver18** [comment]
The comment at the bottom of the path creation block is an optional one liner. I read somewhere that writing raw paths in strings is bad juju in terms of style/guidelines. Even though this particular piece of code should only fire on linux distros, I thought it best to use the idiomatic approach.

File: dex/debugger/lldb/LLDB.py [hunk 1]
```diff
@@ -85,9 +86,17 @@ def _load_interface(self):
...
-            raise LoadDebuggerException(
-                'path "{}" does not exist [result of {}]'.format(
-                    pythonpath, args), sys.exc_info())
+            if "Ubuntu" in platform.linux_distribution():
+                pythonpath = os.path.join("/usr", "lib")
+                pythonpath = os.path.join(pythonpath, "llvm-6.0")
+                pythonpath = os.path.join(pythonpath, "lib")
+                pythonpath = os.path.join(pythonpath, "python2.7")
+                pythonpath = os.path.join(pythonpath, "site-packages")
+                # pythonpath = "/usr/lib/llvm-6.0/lib/python2.7/site-packages/"
```
Comment (TomWeaver18) [inline dex/debugger/lldb/LLDB.py]: This line can be removed if we're happy with the block of joins.

**Comment by gregbedwell** [comment]
Thanks for the patch.  For reference, this is in relation to https://bugs.launchpad.net/ubuntu/+source/llvm-toolchain-6.0/+bug/1776208 right?

I've got a few code nits which I'll comment on in a sec, but my main concern is that this is very specific to one specific version of llvm, so if the same issue happens with the llvm-7.0 package, then it's not going to work, and may even silently cause us to silently end up using the 6.0 interface when we think we're using 7.0.

I'm wondering if a better solution would be to accept '--lldb-python-path' as an option in dexter and use that in preference to the one determined by querying the lldb executable if specified.  What do you think?

**Comment by gregbedwell** [comment]
Nits added, but I think I'd rather go with the approach of letting the user override the path from the command line, unless you strongly disagree, in which case we'll need to figure out how to genericise this.

**Comment by TomWeaver18** [comment]
Dropping this PR, it's no longer a blocking issue for me and seems there wasn't much appetite for it (and rightly so I think).

Cheers
Tom W

**Pull request closed without merging**